src/app/static/src/app/components/plots/MapAdjustScale.vue

```diff
@@ -0,0 +1,71 @@
+import {ColourScaleType} from "../../store/colourScales/colourScales";
+<template>
+    <div v-if="show" class="p-3">
+        <div class="form-check">
+            <label class="form-check-label">
+                <input id="type-input-default" class="form-check-input" type="radio" name="scaleType" :value="ColourScaleType.Default"
+                       v-model="colourScale.type" @change="update">
+                Default
+            </label>
+        </div>
+
+        <div class="form-check">
+            <label class="form-check-label">
+                <input id="type-input-custom" class="form-check-input" type="radio" name="scaleType" :value="ColourScaleType.Custom"
+                       v-model="colourScale.type" @change="update">
+                Custom
+            </label>
+        </div>
+        <div class="ml-2">
+            <div class="mt-2">
+                <label for="custom-min-input">Min</label>
+                <input id="custom-min-input" type="number" v-model.number="colourScale.customMin" @change="update"
+                       :disabled="disableCustom">
+            </div>
+            <div class="mt-2">
+                <label for="custom-max-input">Max</label>
+                <input id="custom-max-input" type="number" v-model.number="colourScale.customMax" @change="update"
+                       :disabled="disableCustom">
+            </div>
+        </div>
+
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import {ColourScaleSettings, ColourScaleType} from "../../store/colourScales/colourScales";
+
+    interface Props {
+        show: Boolean,
+        colourScale: ColourScaleSettings
+    }
+
+    interface Computed {
+        disableCustom: Boolean
+    }
+
+    export default Vue.extend<{}, {}, {}, Props>({
```

Comment:
I'm suprised it didn't complain about this type declaration? Missing computed, methods and data!